apps/api/src/database/repositories/agent-repository.ts

```diff
@@ -170,6 +171,86 @@ export async function findAgentCompetitions(
   }
 }
 
+/**
+ * Find competitions for multiple agents (used for user's agent competitions)
+ * @param agentIds Array of agent IDs
+ * @param params Agent competitions parameters
+ */
+export async function findUserAgentCompetitions(
+  agentIds: string[],
+  params: AgentCompetitionsParams,
+) {
+  try {
+    if (agentIds.length === 0) {
+      return {
+        competitions: [],
+        total: 0,
+      };
+    }
+
+    const { status, claimed } = params;
+
+    // Build where conditions
+    const whereConditions = [inArray(competitionAgents.agentId, agentIds)];
+
+    if (status) {
+      whereConditions.push(eq(competitions.status, status));
+    }
+
+    if (claimed) {
+      console.log(
+        "[AgentRepository] attempting to filter by claimed rewards, but NOT IMPLEMENTED",
+      );
+    }
+
+    let query = db
+      .select()
+      .from(competitionAgents)
+      .leftJoin(agents, eq(competitionAgents.agentId, agents.id))
+      .leftJoin(
+        competitions,
+        eq(competitionAgents.competitionId, competitions.id),
+      )
+      .where(and(...whereConditions))
+      .$dynamic();
+
+    if (params.sort) {
+      query = getSort(query, params.sort, agentCompetitionsOrderByFields);
+    }
+
+    query.limit(params.limit).offset(params.offset);
+
+    const results = await query;
```

Comment:
re: the point above, i think this `results` object should have the agents-related info needed to incorporate it in the final response down below